File: code/ClassPrivateMethod.java
class Animal {
  public void eat(String food) {
    if (isMeat(food))
      System.out.println("Om nom nom");
  }

  private boolean isMeat(String food) {
    return food.equals("meat");
  }
}

class ClassPrivateMethod {
  public static void main(String[] args) {
    new Animal().eat("meat");
  }
}

File: code/php/ClassPrivateMethod.php
<?php

class AnimalI
{
    public function eat(string $food)
    {
        if ($this->isMeat($food))
            echo "Om nom nom";
    }

    private function isMeat(string $food) : bool
    {
        return strcmp($food, "meat") == 0;
    }
}

(new AnimalI())->eat("meat");
